File: ocaml/anagram/anagram.ml
open Base

let normalise word  =  
  word |> String.lowercase |> String.to_list |> List.sort ~compare:Char.compare |> String.of_char_list 

let is_anagram word candidate =  
  String.(normalise candidate = normalise word) && String.(String.lowercase word <> String.lowercase candidate)

let anagrams word candidates =  List.filter ~f:(is_anagram word) candidates

File: julia/anagram/anagram.jl
function detect_anagrams(subject::AbstractString, candidates::AbstractArray)
    normalise(s) = s |> lowercase |> collect |> sort
    isanagram(s₁, s₂) = lowercase(s₁) != lowercase(s₂) && normalise(s₁) == normalise(s₂)
    filter(c -> isanagram(subject, c), candidates)    
end

